### PR TITLE
Pull request to merge my FHQ code to Master

### PR DESCRIPTION
**Pull request by sonika-nih**
Pull request to merge my FHQ code to Master

### CODE REVIEW

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature [hunk 1]
```diff
@@ -7,19 +7,21 @@ Feature: Fanconi Eligibility Questionnaire-Screener on the Fanconi Longitudinal
   Scenario: Participant fills the Fanconi screener for all different scenarios
     Given All scenarios are submitted
 
-  @jains @FanconiSubmission @Regression
+  #@jains @FanconiSubmission @Regression
```
Comment (matakevin) [inline src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature:10]: Is there any reason your tags are commented out?  We could just remove or change the tag to run/not run these tests

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature [hunk 2]
```diff
@@ -7,19 +7,21 @@ Feature: Fanconi Eligibility Questionnaire-Screener on the Fanconi Longitudinal
...
   Scenario: Participant fills the Fanconi screener by the Proband for scenario 1
     Given a Participant is on the Fanconi Study login page
-    And logs in Fanconi page via Okta with username "charmsparticipant1@yopmail.com" and password "Charms123"
+   And logs in Fanconi page via Okta with username "charmsras1@yahoo.com" and password "RASTest2022$$"
+#And logs in Fanconi page via Okta with username "stage-2@yopmail.com" and password "Blue2585#"
```
Comment (matakevin) [inline src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature:14]: Are these commented out lines needed?

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature [hunk 3]
```diff
@@ -7,19 +7,21 @@ Feature: Fanconi Eligibility Questionnaire-Screener on the Fanconi Longitudinal
...
     Then fills the Fanconi Eligibility Questionnaire form for scenario1
     Then clicks the submit button
 
-  @jains @FanconiSubmission
+ @jains @FanconiSubmission 
   Scenario: Participant fills the Fanconi screener by the Proband for scenario 2
     Given a Participant is on the Fanconi Study login page
-    And logs in Fanconi page via Okta with username "charmsparticipant2@yopmail.com" and password "Charms123"
-    And clicks the Fanconi Eligibility Questionnaire widget
+    #And logs in Fanconi page via Okta with username "charmsparticipant2@yopmail.com" and password "Charms123"
```
Comment (matakevin) [inline src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature:22]: Are these commented out lines needed?

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature [hunk 4]
```diff
@@ -1,297 +1,311 @@
 Feature: FHQ Form Submission Scenarios
-  Description:This feature file contains scenarios which submit FHQ Screener by Participant or Legal guardian of the participant
+  Description:This feature file contains scenarios which submit FHQ Screener by Participant or Legal guardian of the participant in the portal and assertion of data in the NativeView
 
-  @FHQRegression
-  Scenario Outline: This scenario outline the completion of the FHQ Form
+  #@FHQSubmissionInPortal @TestRegression
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:4]: Please remove the commented Tag or remove the comment. 
Please change TestRegression to Regression tag

File: src/test/java/ServiceNow/CHARMS/StepsImplementation/FHQSubmissionStepsImpl.java [hunk 5]
```diff
@@ -1,111 +1,164 @@
...
-import appsCommon.Utils.ServiceNow_Common_Methods;
 import org.openqa.selenium.By;
 import org.testng.asserts.SoftAssert;
+
 import com.nci.automation.utils.CucumberLogUtils;
 import com.nci.automation.utils.MiscUtils;
 import com.nci.automation.web.CommonUtils;
 import com.nci.automation.web.EnvUtils;
 import com.nci.automation.web.WebDriverUtils;
+
```
Comment (matakevin) [inline src/test/java/ServiceNow/CHARMS/StepsImplementation/FHQSubmissionStepsImpl.java:11]: We can remove these blank lines at a later PR

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature [hunk 6]
```diff
@@ -1,297 +1,311 @@
...
-      | charmsras1@yahoo.com | RASTest2022$$ | MaternalGrandFather |
-      | charmsras1@yahoo.com | RASTest2022$$ | PaternalGrandMother |
-      | charmsras1@yahoo.com | RASTest2022$$ | PaternalGrandFather |
-      | charmsras1@yahoo.com | RASTest2022$$ | GrandChildren       |
-      | charmsras1@yahoo.com | RASTest2022$$ | Cousin              |
-
-  @FHQRegression @Verified
-  Scenario Outline: This scenario outline the completion of the FHQ Form
+      | Email                | Password      | FHQScenario  | FHQRelative |
+      | charmsras2@yahoo.com | RASTest2023$$ | FHQScenario1 | Participant |
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:22]: It is not recommended to store passwords on the feature file

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature [hunk 7]
```diff
@@ -1,297 +1,311 @@
...
+
+  #@FHQSubmissionInPortal @TestRegression
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:24]: Please either remove the commented tag or remove #. 
Modify TestRegression to Regression tag

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature [hunk 8]
```diff
@@ -1,297 +1,311 @@
...
+
     Examples: 
-      | Email                | Password      | FHQScenario |
-      | charmsras5@yahoo.com | RASTest2023$$ | Partner     |
+      | Email                | Password      | FHQScenario  | FHQRelative |
+      | charmsras2@yahoo.com | RASTest2023$$ | FHQScenario1 | Partner     |
 
-  @FHQRegression @Verified
-  Scenario Outline: This scenario outline the completion of the FHQ Form
+  @FHQSubmissionInPortal
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:35]: Missing Regression tag

File: src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java [hunk 9]
```diff
@@ -10,7 +10,7 @@
 		"pretty", "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"},
 features = "src/test/java/ServiceNow/CHARMS/Features/FHQ",
 glue = "ServiceNow.CHARMS.Steps", 
-tags = "@FHQRegression", 
+tags = "@TestRegression",
```
Comment (matakevin) [inline src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java:13]: This should be the @Regression tag if it is the Regression Runners.  You'll need to change some tags on your feature files for full regression

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature [hunk 10]
```diff
@@ -1,297 +1,311 @@
...
+
     Examples: 
-      | FHQScenario         |
-      | PaternalGrandFather |
+      | FHQScenario  | FHQRelative         |
+      | FHQScenario1 | PaternalGrandFather |
 
-  @FHQRegression
-  Scenario Outline: This scenario outline the assertion of the completed FHQ Form
+  #@FHQAssertionInNV @TestRegression
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:295]: Please either remove the commented tag or remove #.
Modify TestRegression to Regression tag

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature (hunk 3 above)
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature:22]: Please remove the commented lines if no longer needed

File: src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java [hunk 11]
```diff
@@ -10,7 +10,7 @@
 		"pretty", "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"},
 features = "src/test/java/ServiceNow/CHARMS/Features/FHQ",
 glue = "ServiceNow.CHARMS.Steps", 
-tags = "@FHQRegression", 
+tags = "@TestRegression", 
 dryRun = false, 
 monochrome = true,
```
Comment (matakevin) [inline src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java:15]: If you have the latest code from master, you'd have to remove monochrome and strict tags.  We'd also have to remove @RunWith and change the imports to TestNG classes.

File: src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java [hunk 12]
```diff
@@ -1,13 +1,16 @@
 package ServiceNow.CHARMS.Steps;
 
 import java.io.FileNotFoundException;
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java:4]: Please remove the empty line

File: src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java [hunk 13]
```diff
@@ -1,13 +1,16 @@
...
 import com.google.gson.JsonIOException;
 import com.google.gson.JsonSyntaxException;
+import com.nci.automation.utils.CucumberLogUtils;
 import com.nci.automation.utils.MiscUtils;
 import com.nci.automation.web.CommonUtils;
 import com.nci.automation.web.EnvUtils;
 import com.nci.automation.web.WebDriverUtils;
 import com.nci.automation.xceptions.TestingException;
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java:13]: Please remove the empty line

File: src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java [hunk 14]
```diff
@@ -4,6 +4,7 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java:7]: Please remove the empty line

File: src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java [hunk 15]
```diff
@@ -12,12 +13,11 @@
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.ui.Select;
 import org.testng.asserts.SoftAssert;
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java:16]: Please remove the empty line

File: src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java [hunk 16]
```diff
@@ -113,18 +113,35 @@ public static void assertButtonLabelWithSpace(SoftAssert softAssert, WebElement
...
+		String actualValue = webElement.getText();
+		String actualValueWithSpace = actualValue.trim();
+		if (expectedValue == null) {		
+			expectedValue ="";
+		}
+		if(actualValueWithSpace.equals(expectedValue)) {
+			return true;
+		}
+		return false;
+//		softAssert.assertEquals(actualValueWithSpace, expectedValue, "Assertion Failed for" + messsage + "-->");
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java:132]: Please remove the commented line

**Review by matakevin — CHANGES_REQUESTED** [review]
Hey @sonika-nih can you look at my comments on some of these classes.  I'm not too familiar with Charms anymore and the whole Shadow Dom, etc... but please let me know if you have any questions regarding the comments

File: src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSRegressionTest.java [hunk 17]
```diff
@@ -7,7 +7,7 @@
 @CucumberOptions(plugin = {"html:target/html-reports/cucumber-default-report", "json:target/cucumber.json",
 		"junit:target/cucumber.xml", "rerun:target/failed.txt",
 		"pretty", "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"},
-		features = "src/test/java/ServiceNow/CHARMS/Features",
+		features = {"src/test/java/ServiceNow/CHARMS/Features/RasopathyStudy","src/test/java/ServiceNow/CHARMS/Features/FanconiStudy"},
 		glue = "ServiceNow.CHARMS.Steps",
 		tags = "@Regression",
 		dryRun = false,
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSRegressionTest.java:13]: Please use TestNG for your Runner classes

File: src/test/java/appsCommon/Pages/NativeView_SideDoor_Dashboard_Page.java [hunk 18]
```diff
@@ -1,13 +1,15 @@
...
 
-import com.nci.automation.web.WebDriverUtils;
-import io.github.sukgu.Shadow;
-import io.github.sukgu.support.ElementFieldDecorator;
-import io.github.sukgu.support.FindElementBy;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.PageFactory;
 import org.openqa.selenium.support.pagefactory.DefaultElementLocatorFactory;
+
```
Comment (hamidnesar) [inline src/test/java/appsCommon/Pages/NativeView_SideDoor_Dashboard_Page.java:7]: Please remove the empty line

File: src/test/java/appsCommon/Pages/NativeView_SideDoor_Dashboard_Page.java [hunk 19]
```diff
@@ -1,13 +1,15 @@
...
+import com.nci.automation.web.WebDriverUtils;
+
```
Comment (hamidnesar) [inline src/test/java/appsCommon/Pages/NativeView_SideDoor_Dashboard_Page.java:9]: Please remove the empty line

File: src/test/java/appsCommon/Utils/ServiceNow_Common_Methods.java [hunk 20]
```diff
@@ -1,17 +1,17 @@
 package appsCommon.Utils;
 
-import appsCommon.PageInitializers.NativeView_SideDoor_PageInitializer;
-import appsCommon.Pages.NativeViewHomePage;
-import appsCommon.Pages.NativeView_SideDoor_Dashboard_Page;
-import appsCommon.Pages.ServiceNow_NCISP_Page;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.NoSuchElementException;
+
```
Comment (hamidnesar) [inline src/test/java/appsCommon/Utils/ServiceNow_Common_Methods.java:5]: Please remove the empty line

File: src/test/java/appsCommon/Utils/ServiceNow_Common_Methods.java [hunk 21]
```diff
@@ -1,17 +1,17 @@
...
 import com.nci.automation.utils.MiscUtils;
 import com.nci.automation.web.CommonUtils;
 import com.nci.automation.web.EnvUtils;
-import com.nci.automation.web.JavascriptUtils;
 import com.nci.automation.web.WebDriverUtils;
-import org.openqa.selenium.JavascriptExecutor;
-import org.openqa.selenium.Keys;
-import org.openqa.selenium.NoSuchElementException;
+
```
Comment (hamidnesar) [inline src/test/java/appsCommon/Utils/ServiceNow_Common_Methods.java:10]: Please remove the empty line

**Review by hamidnesar — CHANGES_REQUESTED** [review]

File: src/test/java/ServiceNow/CHARMS/Constants/CHARMS_Data_File_Path_Constants.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Constants/CHARMS_Data_File_Path_Constants.java:1]: Please delete empty line 12

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:1]: Do not comment out tags, please delete #

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature:1]: Please do not comment out tags. Also, if scenario is complete, please tag @Regression, if you will organize your scenarios with different tags, please set up a time to discuss

File: src/test/java/ServiceNow/CHARMS/Pages/FHQSubmissionPage.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Pages/FHQSubmissionPage.java:1]: Please delete line 7

File: src/test/java/ServiceNow/CHARMS/Pages/OKTAloginPage.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Pages/OKTAloginPage.java:1]: Please delete line 6

Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Pages/OKTAloginPage.java:1]: Also lines 39,44,45

File: src/test/java/ServiceNow/CHARMS/ScenariosData/FHQ_TestDataManager.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/ScenariosData/FHQ_TestDataManager.java:1]: Please review file and delete empty lines

File: src/test/java/ServiceNow/CHARMS/Steps/FHQSubmissionSteps.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Steps/FHQSubmissionSteps.java:1]: Please delete empty lines

File: src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java:1]: Please delete empty lines and why are you using import com.google.gson.JsonIOException;
import com.google.gson.JsonSyntaxException;?

File: src/test/java/ServiceNow/CHARMS/StepsImplementation/FHQSubmissionStepsImpl.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/StepsImplementation/FHQSubmissionStepsImpl.java:1]: Please delete empty lines

File: src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java:1]: Empty lines again

File: src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java
Comment (iamdez99) [inline src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java:1]: Please set up a time to discuss tags for CHARMS

Comment (iamdez99) [inline src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java:1]: Please update ALL runners classes to run with TestNG - contact me for help if you do not know how

**Review by iamdez99 — CHANGES_REQUESTED** [review]
Please review comments and review all files

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature (hunk 2 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature:14]: Removed the Commented LInes.

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature (hunk 3 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature:22]: Removed the Commented LInes.

File: src/test/java/appsCommon/Utils/ServiceNow_Common_Methods.java (hunk 21 above)
Comment (sonika-nih) [reply src/test/java/appsCommon/Utils/ServiceNow_Common_Methods.java:10]: Done

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:1]: Removed #

File: src/test/java/ServiceNow/CHARMS/Pages/OKTAloginPage.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Pages/OKTAloginPage.java:1]: Done

File: src/test/java/ServiceNow/CHARMS/Pages/FHQSubmissionPage.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Pages/FHQSubmissionPage.java:1]: Done

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature:1]: Removed commented tags

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature (hunk 1 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature]: Commented tags removed

File: src/test/java/ServiceNow/CHARMS/StepsImplementation/FHQSubmissionStepsImpl.java (hunk 5 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/StepsImplementation/FHQSubmissionStepsImpl.java]: Removed blank lines

File: src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java (hunk 11 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java]: Done for all the Runner classes

File: src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java
Comment (matakevin) [inline src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java:1]: Why are you throwing 4 exceptions on line 19?  This step is just to navigate to the Fanconi login page, do you need all of these?  I also think that we're handling the MiscUtils.Sleep exception in the method

File: src/test/java/ServiceNow/ServiceNowRunners/RunSampleRunnerTest.java [hunk 22]
```diff
@@ -1,13 +1,16 @@
...
 import io.cucumber.junit.Cucumber;
 import io.cucumber.junit.CucumberOptions;
-import org.junit.runner.RunWith;
 
 @RunWith(Cucumber.class)
 @CucumberOptions(plugin = { "html:target/html-reports/cucumber-default-report", "json:target/cucumber.json",
         "junit:target/cucumber.xml", "rerun:target/failed.txt",
-        "pretty" }, features = "src/test/java/ServiceNow/SampleTest/Features", glue = "ServiceNow.SampleTest.Steps", tags = "@Sample", dryRun = false, monochrome = false, strict = true)
+        "pretty" }, features = "src/test/java/ServiceNow/SampleTest/Features", glue = "ServiceNow.SampleTest.Steps", tags = "@Sample", dryRun = false, monochrome = false
+        //, strict = true
```
Comment (matakevin) [inline src/test/java/ServiceNow/ServiceNowRunners/RunSampleRunnerTest.java:12]: This line isn't needed seeing as we aren't using the strict tag

**Review by matakevin — CHANGES_REQUESTED** [review]
Check the 2 classes with reviews being asked for.  Thank you!

File: src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java:1]: Removed com.google.gson.JsonIOException;
import com.google.gson.JsonSyntaxException
Also removed extra lines

Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Steps/FanconiLoginSteps.java:1]: Removed all four Exceptions

File: src/test/java/AnalysisTools/MCAExplorer/Features/MCAExplorer.feature [hunk 23]
```diff
@@ -19,7 +19,7 @@ Feature: MCAExplorer features
     And clicks on submit button
     Then verify that there are 0 rows in the result
 
-  @TC3 @satya  @Smoke @Regression n @NCIATWP-6676 @NCIATWP-6797
+  @TC3 @satya  @Smoke @Regression @NCIATWP-6676 @NCIATWP-6797
```
Comment (hamidnesar) [inline src/test/java/AnalysisTools/MCAExplorer/Features/MCAExplorer.feature:22]: Please remove @smoke tag from all scenarios on this file

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQChildRecordProducer.feature [hunk 24]
```diff
@@ -56,7 +56,7 @@ Feature: All FHQ Child Record Producers submisiion and verification in the FHQ S
     And user clicks the Children Grid link on FHQ Grid page to land on the FHQ Childs list view page
     And click the new Child added and verify the completed record producer data
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQChildRecordProducer.feature:57]: All scenarios are missing @Regression tag

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature [hunk 25]
```diff
@@ -10,79 +9,79 @@ Feature: Fanconi Eligibility Questionnaire-Screener on the Fanconi Longitudinal
   @jains @FanconiSubmission @Regression
   Scenario: Participant fills the Fanconi screener by the Proband for scenario 1
     Given a Participant is on the Fanconi Study login page
-    And logs in Fanconi page via Okta with username "charmsparticipant1@yopmail.com" and password "Charms123"
+    And logs in Fanconi page via Okta with username "charmsras1@yahoo.com" and password "RASTest2022$$"
     And clicks the Fanconi Eligibility Questionnaire widget
     Then fills the Fanconi Eligibility Questionnaire form for scenario1
     Then clicks the submit button
 
   @jains @FanconiSubmission
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature:17]: Missing @Regression tag across almost all scenarios

File: src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSTestAccountResetTest.java [hunk 26]
```diff
@@ -10,11 +10,8 @@
 		features = "src/test/java/ServiceNow/CHARMS/Features",
 		glue = "ServiceNow.CHARMS.Steps",
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSTestAccountResetTest.java:11]: This Runner needs to be updated with TestNG

File: src/test/java/ServiceNow/ServiceNowRunners/RunGDCProgressionTest.java [hunk 27]
```diff
@@ -1,25 +1,16 @@
...
-        strict = true
+        dryRun = false
 )
-
-/**
- * This class is annotated with @RunWith(Cucumber.class) and it will run
- * cucumber feature(s).
- */
-public class RunGDCProgressionTest {
+public class RunGDCProgressionTest extends AbstractTestNGCucumberTests{
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/ServiceNowRunners/RunGDCProgressionTest.java:15]: Please remove empty line

**Review by hamidnesar — CHANGES_REQUESTED** [review]
Hi Sonika, I left a few comments on some files if you could please address them. Thank you!

File: src/test/java/AnalysisTools/MCAExplorer/Features/MCAExplorer.feature (hunk 23 above)
Comment (sonika-nih) [reply src/test/java/AnalysisTools/MCAExplorer/Features/MCAExplorer.feature:22]: This is not my code so will not be able to change it

File: src/test/java/AnalysisTools/AnalysisToolsRunners/RunJPSurvProgressionTest.java [hunk 28]
```diff
@@ -1,35 +1,23 @@
...
-
+import org.testng.annotations.BeforeClass;
 import com.nci.automation.utils.LocalConfUtils;
 import com.nci.automation.web.ConfUtils;
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
 
-
-@RunWith(Cucumber.class)
 @CucumberOptions(plugin= {"com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"}
```
[the remark this answers is not in the document]
Comment (sonika-nih) [reply src/test/java/AnalysisTools/AnalysisToolsRunners/RunJPSurvProgressionTest.java:10]: This is not my code so will not be able to change it

File: src/test/java/ServiceNow/ServiceNowRunners/RunGDCProgressionTest.java (hunk 27 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/ServiceNowRunners/RunGDCProgressionTest.java:15]: done

File: src/test/java/AnalysisTools/MCAExplorer/Features/MCAExplorer.feature (hunk 23 above)
Comment (sonika-nih) [reply src/test/java/AnalysisTools/MCAExplorer/Features/MCAExplorer.feature:22]: I removed the @smoke tags even though this is not my code.

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQChildRecordProducer.feature (hunk 24 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQChildRecordProducer.feature:57]: Done

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature (hunk 4 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature]: Done

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature (hunk 6 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature]: I will add this new code later. Need to merge the code for other urgent requirements

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature (hunk 10 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature]: Done

File: src/test/java/ServiceNow/CHARMS/Constants/CHARMS_Data_File_Path_Constants.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Constants/CHARMS_Data_File_Path_Constants.java:1]: Done

File: src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java:1]: Added appropriate tags.

File: src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Utils/CharmsUtil.java:1]: Removed

File: src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSFHQRegressionTest.java:1]: Done

File: src/test/java/ServiceNow/ServiceNowRunners/RunSampleRunnerTest.java (hunk 22 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/ServiceNowRunners/RunSampleRunnerTest.java]: Removed

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature (hunk 7 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature]: Done

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature (hunk 8 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature]: Done

File: src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSRegressionTest.java (hunk 17 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/ServiceNowRunners/RunCHARMSRegressionTest.java]: Done

File: src/test/java/appsCommon/Pages/NativeView_SideDoor_Dashboard_Page.java (hunk 18 above)
Comment (sonika-nih) [reply src/test/java/appsCommon/Pages/NativeView_SideDoor_Dashboard_Page.java]: done

File: src/test/java/appsCommon/Pages/NativeView_SideDoor_Dashboard_Page.java (hunk 19 above)
Comment (sonika-nih) [reply src/test/java/appsCommon/Pages/NativeView_SideDoor_Dashboard_Page.java]: done

File: src/test/java/appsCommon/Utils/ServiceNow_Common_Methods.java (hunk 20 above)
Comment (sonika-nih) [reply src/test/java/appsCommon/Utils/ServiceNow_Common_Methods.java]: Done

File: src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature (hunk 25 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FanconiStudy/FanconiEligibilityQuestionnaire.feature]: done

**Review by hamidnesar — APPROVED** [review]

**Review by iamdez99 — APPROVED** [review]
Reviewed and approved - juarezds

**Pull request merged**